tests/test_config_parsing/test_parser_error_collection.py

```diff
@@ -0,0 +1,221 @@
+import os
+import stat
+from dataclasses import dataclass
+from typing import Callable, Dict, List, Optional, TextIO, Union
+
+import pytest
+
+from ert._c_wrappers.enkf import ErtConfig
+from ert.parsing import ConfigValidationError
+
+test_config_file_base = "test"
+test_config_filename = f"{test_config_file_base}.ert"
+
+
+@dataclass
+class FileDetail:
+    contents: str
+    is_executable: bool = False
+
+
+def write_files(
+    files: Union[dict, FileDetail], write_after: Callable[[TextIO], None] = None
+):
+    if files is not None:
+        for other_filename, content in files.items():
+            if isinstance(content, str):
```

Comment:
The type seems to indicate that content is always FileDetail, also it seems that this function is never called with str